File: LeetCode/Medium/0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        set<char> result;
        int left = 0;
        int right = 0;
        int maxLen = 0;
        
        while(right < s.size()){
            if(result.find(s[right]) != result.end()){
                result.erase(s[left]);
                left += 1;
            }
            else{
                result.insert(s[right]);
                maxLen = max(right - left + 1, maxLen);
                right += 1;
            }
        }
        return maxLen;
        
    }
};

File: LeetCode/Medium/0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        char_set: set = set()
        left: int = 0
        max_len: int = 0

        for right, character in enumerate(s):
            while character in char_set:
                char_set.remove(s[left])
                left += 1
            char_set.add(character)
            max_len = max(max_len, right - left + 1)
        return max_len
        